### PR TITLE
np.atleast_1d() fix for conf_int(method='bca') on scaler inputs.

### LINKED ISSUE #193: Invalid indexer for conf_int() with method='bca' for version 4.2
Hi,

Thank you for this great package. I'm using version 4.2 with Python 3.6. 

I encountered the following error with `conf_int(method='bca')`. 

Looking at the error message & debug info, I wonder if it's just a matter of changing line 463 to:

    a = a[:,]

Happy to create a PR. 

Thank v much.

Code example:
```
import numpy as np
import arch.bootstrap as ab

data = np.random.randn(1000)
sb = ab.StationaryBootstrap(5, data)
ci = sb.conf_int(np.mean, reps=1000, method='bca')

# output:

---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-322-57d5b553d4ae> in <module>()
      2 
      3 sb = arch.bootstrap.StationaryBootstrap(5, data)
----> 4 ci = sb.conf_int(np.mean, reps=1000, method='bca')

H:\Anaconda3\lib\site-packages\arch-4.0-py3.6-win-amd64.egg\arch\bootstrap\base.py in conf_int(self, func, reps, method, size, tail, extra_kwargs, reuse, sampling, std_err_func, studentize_reps)
    461                         raise RuntimeError(message.format(jk_var=denom))
    462                     a = numer / denom
--> 463                     a = a[:, None]
    464                 else:
    465                     a = 0.0

IndexError: invalid index to scalar variable.
```

Here is the debug output:

```
> h:\anaconda3\lib\site-packages\arch-4.0-py3.6-win-amd64.egg\arch\bootstrap\base.py(463)conf_int()
    461                         raise RuntimeError(message.format(jk_var=denom))
    462                     a = numer / denom
--> 463                     a = a[:, None]
    464                 else:
    465                     a = 0.0

ipdb> a
self = Stationary Bootstrap(block size: 5, no. pos. inputs: 1, no. keyword inputs: 0, ID: 0x34501668)
func = <function mean at 0x00000000053340D0>
reps = 1000
method = 'bca'
size = 0.95
tail = 'two'
extra_kwargs = None
reuse = False
sampling = 'nonparametric'
std_err_func = None
studentize_reps = 0
ipdb> a[:,]
self = Stationary Bootstrap(block size: 5, no. pos. inputs: 1, no. keyword inputs: 0, ID: 0x34501668)
func = <function mean at 0x00000000053340D0>
reps = 1000
method = 'bca'
size = 0.95
tail = 'two'
extra_kwargs = None
reuse = False
sampling = 'nonparametric'
std_err_func = None
studentize_reps = 0
ipdb> q
```

### PR DESCRIPTION
**Pull request by esvhd**
This should resolve #193 with `np.atleast_1d()`. I tested the case listed in the issue and it worked. 

Let me know what you think.

Thanks.

### CODE REVIEW

File: arch/bootstrap/base.py [hunk 1]
```diff
@@ -461,7 +461,7 @@ def conf_int(self, func, reps=1000, method='basic', size=0.95, tail='two',
                                   'too small to use BCa'
                         raise RuntimeError(message.format(jk_var=denom))
                     a = numer / denom
-                    a = a[:, None]
+                    a = np.atleast_1d(a)
```
Comment (bashtage) [inline arch/bootstrap/base.py]: I think you need both atleast_1d and [:,None].

Comment (esvhd) [reply arch/bootstrap/base.py]: Tried both `np.atleast_1d(a)` and `np.atleast_1d(a)[:,]`, they produced the same result. [:,None] won't work as it caused the original error. 

Or am I missing something?

See debug output:

```
In [2]: %debug
> /usr/anaconda3/lib/python3.6/site-packages/arch/bootstrap/base.py(464)conf_int()
    462                         raise RuntimeError(message.format(jk_var=denom))
    463                     a = numer / denom
--> 464                     a = a[:, None]
    465                 else:
    466                     a = 0.0
IndexError: invalid index to scalar variable.

ipdb> a
self = Stationary Bootstrap(block size: 5, no. pos. inputs: 1, no. keyword inputs: 0, ID: 0x7fb0e04f8a90)
func = <function mean at 0x7fb0d8306b70>
reps = 1000
method = 'bca'
size = 0.95
tail = 'two'
extra_kwargs = None
reuse = False
sampling = 'nonparametric'
std_err_func = None
studentize_reps = 0
ipdb> np.atleast_1d(a)[:,]
array([ 0.00075776])
ipdb> np.atleast_1d(a)
array([ 0.00075776])
```

**Comment by bashtage** [comment]
It needs both.  The bug happened because `a` was a scalar in your case which cannot be manipulated to 2-d using `[:,None]`.  Calling `atleast_1d` first will ensure that `a` is an array so that calling `[:,None]` second will work.

**Comment by esvhd** [comment]
I see what you mean, will put a new fix in tonight. 

Thx

**Comment by bashtage** [comment]
Looking good.  Ideally this PR would add a test that would have failed under the old version but passes now.  You could do a version of your stat that failed if you feel like it.  If not, I can add after.

**Comment by esvhd** [comment]
Let me take a look at this sometime this weekend.

**Comment by esvhd** [comment]
How about adding this in `tests/bootstrap/test_bootstrap.py`? Asserts output shape and also fails for `IndexError`.

```
    def test_conf_int_bca_scaler(self):
        num_bootstrap = 100
        bs = IIDBootstrap(self.y)
        bs.seed(23456)

        try:
            ci = bs.conf_int(np.mean, reps=num_bootstrap, method='bca')
            assert(ci.shape == (2, 1))
        except IndexError:
            pytest.fail('conf_int(method=\'bca\') scaler input regression. '
                        'Ensure output is at least 1D with '
                        'numpy.atleast_1d().')
```
This passes in the fixed version, but fails in the old version. Tested in both cases.

**Comment by bashtage** [comment]
LGTM.  Hard to not hurt coverage with a test like this.

**Comment by esvhd** [comment]
Great!

**Pull request merged**